File: homework1/38.count_and_say.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
using namespace std;


class Solution {
public:
    string result = "1";
    string countAndSay(int n) {
        if ( n == 1) return "1";
        int count = 1;
        while( count < n){
            result  = RLE(result);
            count++;
        }
        return result;
    }

    string RLE(string str){
        string test = "";
        for ( int i = 0; i< str.length(); i++){
            char s = str[i];
            int count  = 0;
            for ( int j = i; j < str.length(); j++){
                if ( s != str[j]) {
                    break;
                } 
                count++;
            }
            test = test + to_string(count) + s;
            i += count-1  ;
        }
        return test;
    }
};

int main(){
    Solution sol;
    cout << sol.RLE("1144455551") << endl;  
    cout << sol.countAndSay(4) << endl;
}

File: homework1/38_count_and_say.py
class Solution:
    def __init__(self):
        self.result = "1"

    def countAndSay(self, n: int) -> str:
        if n== 1 : return "1"
        count  = 1


        while count < n:
            result  = self.RLE(result)
            count+=1
        return result

    def RLE(self,  strs):
        test = ""
        for i in range(0, len(strs)):
            s = strs[i]
            count = 0
            for j in range(i , len(strs)):
                if s!= strs[j]:
                    break
                count+=1
            test = test + str(count) + s
            i += count - 1
        return test